pkg/configs/advisory/v2/advisory.go

```diff
@@ -111,6 +111,42 @@ func (adv Advisory) ResolvedAtVersion(version string) bool {
 	}
 }
 
+// PendingAtVersion returns true if the advisory indicates that the
+// vulnerability affects the distro package at the given package
+// version, or that a change to the package has not brought in a fix to a CVE.
+func (adv Advisory) PendingAtVersion(version string) bool {
+	if len(adv.Events) == 0 {
+		return true
+	}
+
+	switch latest := adv.Latest(); latest.Type {
+	case EventTypeFalsePositiveDetermination,
+		EventTypeFixNotPlanned,
+		EventTypeAnalysisNotPlanned:
+		return false
+
+	case EventTypeFixed:
+		givenVersion, err := versions.NewVersion(version)
+		if err != nil {
+			return true
+		}
+		fixedData, ok := latest.Data.(Fixed)
+		if !ok {
+			return true
+		}
+		fixedVersion, err := versions.NewVersion(fixedData.FixedVersion)
+		if err != nil {
+			return true
+		}
+
+		fixedInLatest := givenVersion.GreaterThanOrEqual(fixedVersion)
+		return !fixedInLatest
+
+	default:
+		return true
+	}
+}
```

Comment:
This looks like for the most part, it's duplicating logic from the function above, `ResolvedAtVersion`. I think it might be nice at this point to factor out a meaningful common piece of the logic. E.g., we could factor out a function that performs the version-dependent check for `EventTypeFixed`, plus a check against another given set of event types that the caller can decide, which will be used to factor **_out_** the given result.

Reply:
I reused the ResolvedVersion logic as part of the PendingAtVersion, so we don't duplicate the code.